src/components/Drawer.jsx

```diff
@@ -162,37 +157,50 @@ const SideMenuDrawer = React.memo(function SideMenuDrawer({ handleDrawerClose, d
-              <Divider />
-              {items?.at(1)?.map((item, menuIndex) => (
-                <>
-                  <ListItem
-                    key={item.name}
-                    button
-                    onClick={() => {
-                      if (item.sub_menu.length > 0 && !isSubMenOpen[typeIndex][menuIndex]) {
-                        handleSubMenOpenClick(typeIndex, menuIndex);
-                      }
-                      if (item.sub_menu.length == 0) {
-                        navigate(item.url);
-                      }
-                    }}
-                  >
-                    <ListItemIcon>{item.icon}</ListItemIcon>
-                    <ListItemText primary={item.name} />
-
-                    {item.sub_menu && showMoreOrLessIcon(item.sub_menu, typeIndex, menuIndex)}
-                  </ListItem>
-                  {item.sub_menu && renderSubMenu(item.sub_menu, typeIndex, menuIndex)}
-                </>
-              ))}
-            </Fragment>
-          ))}
-      </List>
+
+      {/* Added container with fixed height and overflow */}
+      <Box style={{ overflow: "auto", maxHeight: "calc(100vh - 64px)" }}>
+        <List style={{ padding: "2px 0" }}>
+          {groupedMenu &&
+            Object.entries(groupedMenu).map(([type, items], typeIndex) => (
+              <Fragment key={type}>
+                <ListSubheader sx={{ fontWeight: "bold" }}>{type === "public" ? "Public" : "Secure"}</ListSubheader>
+                <Divider sx={{ marginBottom: 1 }} />
+                {items.map((item, menuIndex) => (
+                  <Fragment key={item.name}>
+                    <ListItem
+                      button
+                      dense
+                      onClick={() => {
+                        if (item.sub_menu.length > 0 && !isSubMenOpen[typeIndex][menuIndex]) {
+                          handleSubMenOpenClick(typeIndex, menuIndex);
+                        }
+                        if (item.sub_menu.length === 0) {
+                          navigate(item.url);
+                        }
+                      }}
+                      sx={{ py: 1 }} // Adjusted vertical padding
+                    >
+                      <ListItemIcon>{item.icon}</ListItemIcon>
+                      <ListItemText primary={item.name} />
+                      {item.sub_menu && showMoreOrLessIcon(item.sub_menu, typeIndex, menuIndex)}
+                    </ListItem>
+                    {item.sub_menu && renderSubMenu(item.sub_menu, typeIndex, menuIndex)}
+                  </Fragment>
+                ))}
+              </Fragment>
+            ))}
+          {!isAuthenticated ? (
+            <StyledCustomButtons onClick={() => handleNavigation("/login")} color="primary">
+              Login
+            </StyledCustomButtons>
+          ) : (
+            <StyledCustomButtons onClick={() => handleNavigation("/logout")} color="primary">
```

Comment:
I don't think we will have a logout page to redirect to
So here you can pass the callType like login or logout and in handleNavigation function you can have a switch case for the received type and perform need action form there

Reply:
User will be redirected to login page on logout button click or to the public home page